server/cssfile.js

```diff
@@ -35,42 +35,64 @@ CssFile.prototype.selectorFromPosition = function(line, column){
 };
 
 CssFile.prototype.setContent = function(source, callback){
-	var messages = csslint.verify(source).messages;
-	var errors = [];
-	messages.forEach(function(msg){
-		if(msg.type == 'error'){
-			errors.push(msg);
+	const errors = [];
+	this.source = source;
+	const lint = stylelint.lint({code: source}).then((result) => result.results).then(results => {
+		results.forEach((msg) =>{
+			if(msg.errored){
+				msg.warnings.forEach(warning => {
+					errors.push(warning)
+				});
+			}
+		})
+		return results
+	}).then(results => {
+		if(errors.length > 0 && callback){
+			throw new Error("Errors found")
 		}
-	});
+		return results
+	}).catch(err => {
+		return err
+	})
 
-	if(errors.length > 0 && callback){
-		callback(errors);
-		return;
+	const parser =  new Promise((res, rej) => { 
+		try {
+			const parsed = cssparser.parse(source)
```

Comment:
since cssparser is already a simple blocking call I don't think it should be wrapped in a promise.

Reply:
I tried not wrapping it in a promise, but because stylelint uses a promise.

So, if I don't use a promise in the cssparsed, it'll always try to parse the code, even with lint giving an error.

This would raise exceptions, because it would try to parse a wrong css code.

Putting it in an promise delay its execution, and allow stylelint to error, having a callback with the errors, as it was before

Reply:
My bad, I should have been more clear. You can just call `cssparser.parse` inside the `.then` callback, There's no need to create the second promise. This also has the advantage of not running the css parser when you don't have to.

Reply:
I tried this, but because of the `this.parsed = cssparser.parse(source)` and the `this.parsed` call in selectorFromPosition function, using `cssparser.parse` in the `.then` would give an error, because `this.parsed` would be `undefined`

Reply:
I'm having a bit of trouble understanding why `this.parsed` being undefined would be an error. It shouldn't matter if `this.parsed` is undefined since it's only being set. It might be that perhaps `this` is undefined if you're not using an arrow function???

Reply:
No, the error is in the function `selectorFromPosition`, because it uses `this.parsed` in its for loop, so if it's undefined, it'll raise an exception

Putting the cssparsed in the `.then` would make delay its execution, making the `selectedFromPosition` to try to interact with an undefined variable

But when I put the css parsed in its own promise, it resolves this problem

Reply:
Ah gotcha! It'd be best to match the previous behavior which leaves `this.parsed` undefined when there is a lint or parse error. As long as `setContent` matches the previous behavior we shouldn't have to worry about it. The previous behavior was to leave `this.parsed` undefined so lets do that that! For reference I believe the previous goal of the linter was as a safeguard the parse function, I'm suspicious the linting rules may have accidentally changed when moving to the new linter.

Side note: Sounds like `selectorFromPosition` erring when `this.parsed` is undefined is a separate bug we can address in another PR. As long as we match the previous behavior this shouldn't matter.